comp/otelcol/otlp/components/logsagentpipeline/agent.go

```diff
@@ -0,0 +1,192 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+package logsagentpipeline
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	configComponent "github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/hostname/hostnameinterface"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	logComponent "github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/logs/agent/config"
+	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
+	"github.com/DataDog/datadog-agent/pkg/logs/client"
+	"github.com/DataDog/datadog-agent/pkg/logs/pipeline"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+	"github.com/DataDog/datadog-agent/pkg/util/startstop"
+
+	"go.uber.org/fx"
+	"go.uber.org/zap"
+)
+
+const (
+	intakeTrackType = "logs"
+
+	// Log messages
+	multiLineWarning = "multi_line processing rules are not supported as global processing rules."
+)
+
+type dependencies struct {
+	fx.In
+
+	Lc       fx.Lifecycle
+	Log      logComponent.Component
+	Config   configComponent.Component
+	Hostname hostnameinterface.Component
+}
+
+type provides struct {
+	fx.Out
+
+	Comp optional.Option[Component]
+}
+
+// Agent represents the data pipeline that collects, decodes, processes and sends logs to the backend.
+type Agent struct {
+	log      log.Component
+	config   pkgconfigmodel.Reader
+	hostname hostnameinterface.Component
+
+	endpoints        *config.Endpoints
+	auditor          auditor.Auditor
+	destinationsCtx  *client.DestinationsContext
+	pipelineProvider pipeline.Provider
+	health           *health.Handle
+}
+
+func NewLogsAgent(deps dependencies) provides {
+	if deps.Config.GetBool("logs_enabled") || deps.Config.GetBool("log_enabled") {
+		if deps.Config.GetBool("log_enabled") {
+			deps.Log.Warn(`"log_enabled" is deprecated, use "logs_enabled" instead`)
+		}
+
+		logsAgent := &Agent{
+			log:      deps.Log,
+			config:   deps.Config,
+			hostname: deps.Hostname,
+		}
+		if deps.Lc != nil {
+			deps.Lc.Append(fx.Hook{
+				OnStart: logsAgent.Start,
+				OnStop:  logsAgent.Stop,
+			})
+		}
+
+		return provides{
+			Comp: optional.NewOption[Component](logsAgent),
+		}
+	}
+
+	deps.Log.Debug("logs-agent disabled")
+	return provides{
```

Comment:
NIT: if you are only returning the component - you can just return `optional.NewNoneOption[Component]()` directly instead of using `provides`. provides is useful if you are fulfilling multiple dependences.